### PR TITLE
Add ability to attach layout handles to page types

### LINKED ISSUE #11: FR: Ability to add layout handles to page types
Currently you are able to define a single template to render the content and fields added by TypeCMS.

There may be some instances in which you require more control over the layout of a specific page type. E.g. you may want to force the template to `2column-left` view and add a submenu in the left column for all pages with that type.

To do this the page would require a layout handle adding then using Magento's layout XML in a theme you can manipulate the page layout as a whole. 

The only alternative is to create page layouts in an additional module `config.xml` like so: 

```xml
    <global>
        <page>
            <layouts>
                <custom_layout translate="label">
                    <label>Custom Layout Label</label>
                    <template>page/1column.phtml</template>
                    <layout_handle>custom_layout_handle</layout_handle>
                </custom_layout>
            </layouts>            
        </page>
    </global>
```

### PR DESCRIPTION
**Pull request by adamj88**
Fixes #11

### CODE REVIEW

**Comment by adamj88** [comment]
There's a repetition of `$node = Mage::getConfig()->getNode(self::XML_PATH_PAGE_TYPES)->descend($pageType);` in https://github.com/adamj88/RK_TypeCMS/blob/b0a07b40ea121e68d70f79496fcc751f23d161a6/app/code/community/RK/TypeCMS/Model/Config.php (lines 42 & 62).

Would it be best to refactor this into an additional function?

File: app/code/community/RK/TypeCMS/Model/Observer.php [hunk 1]
```diff
@@ -117,4 +117,17 @@ protected static function deleteImage($file)
...
     }
 
+    /**
+     * @param  Varien_Event_Observer
+     */
+    public function onCmsRenderPage(Varien_Event_Observer $observer)
+    {
+        $event = $observer->getEvent();
+        $page = $event->getPage();
+        if($handle = Mage::getModel('typecms/page')->load($page->getId())->getPageTypeInstance()->getHandle()) {
```
Comment (rskuipers) [inline app/code/community/RK/TypeCMS/Model/Observer.php]: Can you add a space here between the `if` and the expression?

**Comment by rskuipers** [comment]
Awesome! I have one code comment and one functional suggestion.
Can you give the `default` and `collection` page types in `app\code\community\RK\TypeCMS\etc\config.xml` a default handle? So for `default` we can do `typecms_page` and for collection we can do `typecms_collection`.

Great job on the PR, once these two small additions are done I will merge and release a new version :)

**Comment by adamj88** [comment]
@rskuipers sorry didn't see this message. We're still using typecms here!

**Comment by rskuipers** [comment]
@adamj88 That's good to hear Adam, thank you for your contribution :+1:

**Pull request merged**